### ISSUE TITLE
user key = clear user passphrase + hash + secret transformation algorithm

### ISSUE THREAD

**Issue by danielweck** [readium/readium-lcp-server#136]
Question: is there some LCP server code that consumes a clear passphrase (to generate a fresh new license, or to retrieve / re-generate existing license data), or is the passphrase always first hashed on the calling side (e.g. the frontend proxy between the user and the LCP server)?

**Comment by danielweck** [comment]
Code references:
https://github.com/readium/readium-lcp-server/blob/develop/lcpserver/api/license.go#L335
`License.Encryption.UserKey.ClearValue`
https://github.com/readium/readium-lcp-server/blob/develop/lcpserver/api/license.go#L481
```go
		passphrase := l.Encryption.UserKey.ClearValue
		l.Encryption.UserKey.ClearValue = ""
		hash := sha256.Sum256([]byte(passphrase))
		encryptionKey = hash[:]
```

**Comment by danielweck** [comment]
The old LCP server "manage" code is obsolete, let's remove it to avoid confusion (it used `UserKey.ClearValue`):
https://github.com/readium/readium-lcp-server/blob/develop/lcpserver/manage/index.html#L115
`  obj.encryption.user_key.clear_value = $('#passphrase').val();`

**Comment by danielweck** [comment]
If we deprecate / prevent the use of `UserKey.ClearValue`, then let's remove it from the data model too:
https://github.com/readium/readium-lcp-server/blob/develop/license/license.go#L52
`ClearValue string `json:"clear_value,omitempty"`

**Comment by danielweck** [comment]
Finally, I believe that the "partial license" model in Javascript (well, Typescript) is not used anymore, right @clebeaupin ( @snemegeer ):
https://github.com/readium/readium-lcp-server/blob/develop/frontend/manage/app/components/partialLicense.ts#L21
`    clear_value: string | undefined;`

**Comment by snemegeer** [comment]
I do like the lcpserver manage code, because it connects directly to the lcp server (not via purchases, etc)
for example : the listing of licenses etc is not included in the frontend code.

**Comment by llemeurfr** [comment]
Why not completing the new frontend code then by adding the list of licenses?

**Comment by snemegeer** [comment]
Yes, that would be fine (we could then also add a method to list overuse of licenses by checking the lsd server ) .  For example a top 10 of most registered licenses and the option to revoke the license.
* license xyz1: 6234 registered devices  [revoke license]
* license xyz2: 214 registered devices  [revoke license]

**Comment by danielweck** [comment]
also, we can keep lcp manage/index.html and just migrate from clear to hashed passphrase (which would allow us to completely remove user_key.clear_value).

**Comment by llemeurfr** [comment]
Fixed in 362aaeb0e788eec1e74969febb86208984da1b66.

The frontend lists licenses with a query on the minimum number of devices registered for the license.
